### PR TITLE
chore: switch to PM

### LINKED ISSUE #1465: request: replace RichDisplay with PaginatedMessage
**Is your feature request related to a problem? Please describe.**

RichDisplay is a Klasa-specific thing, but not only that, it's also flawed in many ways.

**Describe the solution you'd like**

[`@sapphire/discord.js-utilities`](https://www.npmjs.com/package/@sapphire/discord.js-utilities) has `PaginatedMessage`, which has all the features we need.

**Describe alternatives you've considered**

None.

**Additional context**

N/a.

### PR DESCRIPTION
**Pull request by kyranet**
Supersedes #1488
Fixes #1465

### CODE REVIEW

File: src/commands/Management/triggers.ts [hunk 1]
```diff
@@ -119,17 +119,18 @@ export default class extends SkyraCommand {
...
+		});
 
 		for (const page of chunk(output, 10)) {
-			display.addPage((embed: MessageEmbed) => embed.setDescription(page));
+			display.addPageEmbed((embed) => embed.setDescription(page));
 		}
 
-		return display.start(message, undefined, { time: 120000 });
+		display.setIdle(120000);
+		return display.start(message);
```
Comment (vladfrangu) [inline src/commands/Management/triggers.ts]: Use Time enum, or this
```suggestion
		return display.setIdle(120_000).start(message);
```

File: src/lib/structures/UserPaginatedMessage.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	/**
+	 * Sets up the message's reactions and the collector.
+	 * @param channel The channel the handler is running at.
+	 * @param author The author the handler is for.
+	 */
+	protected async setUpReactions(channel: TextChannel | NewsChannel, author: User): Promise<void> {
+		if (this.pages.length > 1) await super.setUpReactions(channel, author);
+	}
```
Comment (vladfrangu) [inline src/lib/structures/UserPaginatedMessage.ts:64]: Can't this be merged in upstream?

File: src/lib/structures/UserPaginatedMessage.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+			video: embed.video ?? template.video ?? undefined,
+			footer: embed.footer ?? template.footer ?? undefined
+		};
+	}
+
+	private mergeArrays<T>(template?: T[], array?: T[]): undefined | T[] {
+		if (!array) return template;
+		if (!template) return array;
+		return [...template, ...array];
+	}
```
Comment (vladfrangu) [inline src/lib/structures/UserPaginatedMessage.ts:118]: probably best as a util

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/lib/structures/UserPaginatedMessage.ts (hunk 2 above)
Comment (kyranet) [reply src/lib/structures/UserPaginatedMessage.ts:64]: wdym?

File: src/lib/structures/UserPaginatedMessage.ts (hunk 3 above)
Comment (kyranet) [reply src/lib/structures/UserPaginatedMessage.ts:118]: Probably in another PR, I cba to write tests for those utilities, specially with the amount of cases they handle.

File: src/lib/structures/UserPaginatedMessage.ts (hunk 2 above)
Comment (vladfrangu) [reply src/lib/structures/UserPaginatedMessage.ts:64]: aka move the if check in the main module

Comment (kyranet) [reply src/lib/structures/UserPaginatedMessage.ts:64]: If that behaviour is desirable, I can move it upstream.

However... what if somebody defines custom actions which do page-independent actions? Say an info page.

Comment (vladfrangu) [reply src/lib/structures/UserPaginatedMessage.ts:64]: Well that'd be a different page so.. 🤔

Comment (kyranet) [reply src/lib/structures/UserPaginatedMessage.ts:64]: Well... nothing stops them from extending the behaviour to add virtual pages which aren't on the `this.pages` array.

Comment (vladfrangu) [reply src/lib/structures/UserPaginatedMessage.ts:64]: Riight, but... what's the point of reacting with one page? imo, if you want that behavior, you WANT to extend anyways

**Review by favna — APPROVED** [review]

**Pull request merged**